src/encoder/serde.rs

```diff
@@ -76,6 +76,7 @@ pub struct Encoder;
 
 impl Encoder {
     /// Construct a new `Serializer`.
+    #[allow(clippy::new_without_default)]
```

Comment:
In general, implementing `new` with no parameters means that you can also implement or derive `Default`. For now, I've just whitelisted each occurrence of this pattern.